docs/writing-stories/args.md

```diff
@@ -207,6 +207,21 @@ Similarly, special formats are available for dates and colors. Date objects will
 
 Args specified through the URL will extend and override any default values of args set on the story.
 
+## Setting args from within a story
+
+Components with interactivity often need their containing component, or page, to respond to events, modify their state, then pass a changed arg back to the rendered component to reflect the change. For example, you would want a Switch component to be checked by a user and the arg shown in Storybook to reflect the change. This can be accomplished by using the `useArgs` hook exported by `@storybook/preview-api`:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/this-pr-example-snippet.js.mdx',
+    'react/page-story-args-within-story.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
```

Comment:
@jonthenerd, what do you think about rephrasing it to:

```md
Interactive components often need to be controlled by their containing component or page to respond to events, modify their state and reflect those changes in the UI. For example, when a user toggles a switch component, the switch should be checked, and the arg shown in Storybook should reflect the change. To enable this, you can use the [`useArgs`](../addons/addons-api.md#useargs) API exported by `@storybook/preview-api`:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'react/page-story-args-within-story.js.mdx', 
    'react/page-story-args-within-story.ts.mdx',
  ]}
/>

<!-- prettier-ignore-end -->
```

Reply:
Made the change to the text as requested. However, the link is to a page describing the `manager-api` and we're talking here about using the `preview-api`. Is that okay?

Reply:
Sorry for the confusion here, but we shouldn't be pointing to an example aimed at the `manager-api` as it will introduce a misconception. If you don't mind, could you remove the link for now? I'll take a more thorough look tomorrow, and if nothing arises we can get this merged in.

Reply:
Figured as much. Removed the link. Ready to go!